### PR TITLE
feat: support lb reusable

### PR DESCRIPTION
**Pull request by adevjoe**
bcs-ingress-controller 支持 lb 复用，创建和更新 `portpool` 时会由 webhook 检测端口冲突。

### CODE REVIEW

File: bcs-runtime/bcs-k8s/bcs-network/bcs-ingress-controller/internal/webhookserver/validate.go [hunk 1]
```diff
@@ -86,13 +89,16 @@ func (s *Server) checkPortPool(newPool *networkextensionv1.PortPool) error {
 			if err != nil {
 				return fmt.Errorf("lbIDStr %s of item %s is invalid", lbIDStr, item.ItemName)
 			}
-			if itemName, ok := lbIDMap[lbID]; ok {
-				return fmt.Errorf("lbID %s of item %s conflicts with id of item %s", lbID, item.ItemName, itemName)
-			}
-			if arn.IsARN(lbID) {
-				isARN = true
+			for port := item.StartPort; port < item.EndPort; port++ {
+				lbPort := fmt.Sprintf("%s-%d", lbID, port)
```
Comment (liuchenyiwork) [inline bcs-runtime/bcs-k8s/bcs-network/bcs-ingress-controller/internal/webhookserver/validate.go:93]: recommend to use the function: common.go/GetListernerName

File: bcs-runtime/bcs-k8s/bcs-network/bcs-ingress-controller/internal/webhookserver/validate.go [hunk 2]
```diff
@@ -159,7 +157,10 @@ func (s *Server) checkPortPoolConflicts(newPool *networkextensionv1.PortPool) er
 				if err != nil {
 					return fmt.Errorf("lbIDStr %s of existed item %s is invalid", item.ItemName, lbIDStr)
 				}
-				lbIDMap[lbID] = existedPool.GetName() + "/" + existedPool.GetNamespace()
+				for port := item.StartPort; port < item.EndPort; port++ {
+					lbPort := fmt.Sprintf("%s-%d", lbID, port)
```
Comment (liuchenyiwork) [inline bcs-runtime/bcs-k8s/bcs-network/bcs-ingress-controller/internal/webhookserver/validate.go:161]: recommend to use the function: common.go/GetListernerName

File: bcs-runtime/bcs-k8s/bcs-network/bcs-ingress-controller/internal/webhookserver/validate.go [hunk 3]
```diff
@@ -170,9 +171,12 @@ func (s *Server) checkPortPoolConflicts(newPool *networkextensionv1.PortPool) er
 			if err != nil {
 				return fmt.Errorf("lbIDStr %s of new item %s is invalid", newItem.ItemName, lbIDStr)
 			}
-			if existedPoolKey, ok := lbIDMap[lbID]; ok {
-				return fmt.Errorf("lbID %s of new item %s is conflict with pool %s",
-					lbID, newItem.ItemName, existedPoolKey)
+			for port := newItem.StartPort; port < newItem.EndPort; port++ {
+				lbPort := fmt.Sprintf("%s-%d", lbID, port)
```
Comment (liuchenyiwork) [inline bcs-runtime/bcs-k8s/bcs-network/bcs-ingress-controller/internal/webhookserver/validate.go:175]: recommend to use the function: common.go/GetListernerName

**Review by liuchenyiwork — COMMENTED** [review]
It is recommended to use uniform functions to optimize the code.

**Review by silenceper — APPROVED** [review]

**Pull request merged**